server/leafnode.go

```diff
@@ -281,10 +283,12 @@ func (s *Server) connectToRemoteLeafNode(remote *leafNodeCfg, firstConnect bool)
 		// Go ahead and create our leaf node and return.
 		s.createLeafNode(conn, remote)
 
-		// We will put this in the normal log if first connect, does not force -DV mode to know
-		// that the connect worked.
+		// We will put this in the normal log if first connect or when leafnode changed,
+		// does not force -DV mode to know that the connect/reconnect worked.
 		if firstConnect {
 			s.Noticef("Connected leafnode to %q", rURL.Hostname())
+		} else if leafURLChanged {
```

Comment:
Not sure... would not be that if you were to reconnect to the same URL you would not see the message? Would it not be simpler to simply remove the `firstConnect` from original code and that's it? Or then simply something like 
```
if firstConnect {
  ..
} else {
  s.Noticef("Reconnected ...
}
```
which means that if not `firstConnect` we are in the situation where the connection was lost and this is a solicited connection so we call this function again.

Reply:
Yes as it is currently it would not report if reconnecting to the same URL, thought we may want to have less logging entries but I agree would be simpler to report the successful connect or reconnect as you mention.

Reply:
We already suppress failed attempts based on some threshold that can be configured. But for the connected log, I would simply have a single Noticef, or based on firstConnect just have the Reconnected mention.